### ISSUE TITLE
can we have Object3D.remove() with no arguments remove an object itself?

### ISSUE THREAD

**Issue by makc** [mrdoob/three.js#21714]
so that we can write `light.remove();` instead of `light.parent.remove(light);` ?

right now .remove() does not seem to be doing anything so there will be no conflict.

**Comment by Mugen87** [comment]
I think this is a good suggestion. I've often encountered issues where this semantic would be convenient. The method would then behave like [the remove() method of DOM elements](https://www.w3schools.com/jsref/met_element_remove.asp) if invocated with no parameters.

**Comment by WestLangley** [comment]
>can we have Object3D.remove() with no arguments remove an object itself?

If `.remove()` had been named `.removeChild()` then you could have.

I would oppose redefining the behavior of the current method.

**Comment by donmccurdy** [comment]
Maybe something like `object.detach()`?

**Comment by mrdoob** [comment]
`object.removeSelf()`?

**Comment by makc** [comment]
`object.suicide()`

but no, this kind of name sounds like remove + dispose

**Comment by mrdoob** [comment]
`object.removeItself()`?

**Comment by makc** [comment]
`object.getLost()`

**Comment by jaeh** [comment]
`object.destroy()`

this would be kinda obvious and be the same name used in nodejs streams (for example)

also can remember some engines using destroy, so, imho, it would be familiar

**Comment by donmccurdy** [comment]
The purpose of the method is to {disconnect, detach, unlink, unparent, remove} an object from the scene graph, without changing or destroying the object itself. Names like `destroy()` sound more like our current `dispose()` method, permanently de-allocating the object's resources.

Blender calls this operation [*Clear Parent*](https://docs.blender.org/manual/en/latest/scene_layout/object/editing/parent.html#clear-parent). jQuery calls it [*.detach()*](https://api.jquery.com/detach/).

**Comment by WestLangley** [comment]
If the the parent is known, then one would do this:
```js
parent.remove( child );
```
If the parent is not known, then one would do this:
```js
parent = child.parent;

if ( parent ) parent.remove( child );
```

I think the current API is fine as-is.

**Comment by makc** [comment]
sufficient - yes. fine? nah.

**Comment by Mugen87** [comment]
Reconsidering: Since there is already `Object3D.attach()`, I vote for `Object3D.detach()`.

The method detaches a 3D object from its parent. To keep it similar to `attach()`, it would ensure the 3D object keeps it current world transform.

**Comment by WestLangley** [comment]
> Since there is already Object3D.attach(), I vote for Object3D.detach()

If we had implemented it as `Object3D.attach( parent )`, meaning "attach to parent",  then `Object3D.detach()`, meaning "detach from parent", would make sense.

**Comment by makc** [comment]
@WestLangley if you are ok with NEW method, and the only problem is the name, just have object.removeFromParent() or something.

**Comment by makc** [comment]
@mrdoob maybe you could catch this by arguments.length === 0? not sure myself

**Comment by donmccurdy** [comment]
I think that ship has sailed, it's too late to use `.remove()` here. I'm fine with adding a new method, but no strong preference on whether it's needed.

**Comment by mrdoob** [comment]
`object.removeFromParent()` sounds like the best option to me.

**Comment by Mugen87** [comment]
Sounds good!
